zipkin-storage/cassandra/src/main/java/zipkin2/storage/cassandra/DefaultSessionFactory.java

```diff
@@ -45,27 +47,23 @@
    */
   @Override
   public Session create(CassandraStorage cassandra) {
-    Cluster cluster = null;
     Session session = null;
     try {
-      cluster = buildCluster(cassandra);
-      cluster.register(new QueryLogger.Builder().build());
+      session = buildSession(cassandra);
+
       String keyspace = cassandra.keyspace();
       if (cassandra.ensureSchema()) {
-        session = cluster.connect();
         Schema.ensureExists(keyspace, cassandra.searchEnabled(), session);
-        session.execute("USE " + keyspace);
       } else {
         LOG.debug("Skipping schema check on keyspace {} as ensureSchema was false", keyspace);
-        session = cluster.connect(keyspace);
       }
 
+      session.execute("USE " + keyspace);
```

Comment:
Assuming connect / use is equivalent to `cluster.connect(keyspace)` from the old else branch

Reply:
yeah I looked into the code. basically cluster.connect(keyspace) is a composition of connect and use